packages/git-encrypted/src/crypto.ts

```diff
@@ -163,3 +167,35 @@ export const decryptFileContentsOnly = async ({
 
   return decryptedContents;
 };
+
+export const ensureKeysExist = async ({
+  fs,
+  gitdir,
+  encryptedDir,
+  encryptedKeysDir,
+  keyDerivationPassword,
+}: Pick<GitBaseParamsEncrypted, 'fs' | 'gitdir'> & {
+  encryptedDir: string;
+  encryptedKeysDir: string;
+  keyDerivationPassword?: string;
+}) => {
+  const encryptedKeysDirectoryExists = await doesDirectoryExist({
+    fs,
+    path: encryptedKeysDir,
+  });
+
+  if (encryptedKeysDirectoryExists) {
+    return;
+  }
+
+  if (typeof keyDerivationPassword === 'undefined') {
+    throw new Error(
+      'Cannot initialise without key derivation password #xvoEqa'
+    );
+  }
+
+  const meta = await ensureMetaExists({ fs, encryptedDir });
+
+  const keys = await deriveKeys({ ...meta.derivationParams, password: keyDerivationPassword });
```

Comment:
nit: Is `...meta.derivationParams` actually the correct API for `deriveKeys`? I guess there's a `dkLen` parameter which we don't save into `meta`, but maybe that's not required...

Again, not sure, just wondering aloud...

Reply:
`deriveKeys` does not expect a key length, it is always 96 bytes. Otherwise it looks like it's the right API, otherwise typescript would also complain?